src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,30 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task{
+    protected LocalDateTime by;
```

Comment:
```suggestion
    protected LocalDateTime dueDate;
```
Perhaps it would be better if you use dueDate instead of by? That could make field sound more intuitive especially when its referenced in other parts of the code.